### PR TITLE
Make SQLServerException constructors public

### PR DESCRIPTION
**Pull request by mhaderman**
Make SQLServerException constructors public so that interfaces like ISQLServerBulk record can be implemented outside of the package.

### CODE REVIEW

**Comment by v-nisidh** [comment]
@mhaderman : Thanks. Can you push code from your dev branch? I just want to avoid cherry-pick merge.

**Comment by v-suhame** [comment]
@mhaderman please remove README file from the commit.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerException.java [hunk 1]
```diff
@@ -132,14 +132,14 @@ static String getErrString(String errCode) {
...
-    SQLServerException(String errText,
+    public SQLServerException(String errText,
             SQLState sqlState,
             DriverError driverError,
             Throwable cause) {
         this(errText, sqlState.getSQLStateCode(), driverError.getErrorCode(), cause);
     }
 
-    SQLServerException(String errText,
+    public SQLServerException(String errText,
```
Comment (v-suhame) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerException.java]: Add javadocs to these new public APIs.

**Comment by v-suhame** [comment]
@mhaderman can you submit this PR using dev as base branch? Right now the PR is using **master** as source and **dev** as destination, so there are additional commits in this PR that are not intended.

**Comment by v-nisidh** [comment]
Closing with ref #247

**Pull request closed without merging**